src/util/address.rs

```diff
@@ -468,6 +468,15 @@ impl Payload {
             program: output_key.as_inner().serialize().to_vec(),
         }
     }
+
+    /// Returns a byte slice of the payload
+    pub fn as_bytes(&self) -> &[u8] {
+        match self {
+            Payload::ScriptHash(hash) => hash,
+            Payload::PubkeyHash(hash) => hash,
+            Payload::WitnessProgram { program, .. } => program,
```

Comment:
Is this even correct? Witness version is normally part of the script and it changes the semantics of the program. Omitting it could lead to serious problems but we can't return it in a slice.

Reply:
This make sense to me as a convenience function. The version is typically handled separately being a u5. The witness program itself without the version is analogous to both the ScriptHash and PubkeyHash that are also returned here. In the case of verifying that a key was used to derive an address, this makes a lot of sense to me.

Reply:
I'm also a bit uncomfortable with this, but I can see both sides of the argument.

Reply:
Perhaps we should add more thorough docs to this method?